src/transformers/models/gpt2/tokenization_gpt2.py

```diff
@@ -261,6 +261,38 @@ def build_inputs_with_special_tokens(self, token_ids_0, token_ids_1=None):
 
         return output + bos_token_ids + token_ids_1
 
+    def get_special_tokens_mask(
+        self, token_ids_0: List[int], token_ids_1: Optional[List[int]] = None, already_has_special_tokens: bool = False
+    ) -> List[int]:
+        """
+        Retrieves sequence ids from a token list that has no special tokens added. This method is called when adding
+        special tokens using the tokenizer `prepare_for_model` or `encode_plus` methods.
+
+        Args:
+            token_ids_0 (`List[int]`):
+                List of IDs.
+            token_ids_1 (`List[int]`, *optional*):
+                Optional second list of IDs for sequence pairs.
+            already_has_special_tokens (`bool`, *optional*, defaults to `False`):
+                Whether or not the token list is already formatted with special tokens for the model.
+
+        Returns:
+            `List[int]`: A list of integers in the range [0, 1]: 1 for a special token, 0 for a sequence token.
+        """
+        if already_has_special_tokens:
+            return super().get_special_tokens_mask(
+                token_ids_0=token_ids_0, token_ids_1=token_ids_1, already_has_special_tokens=True
+            )
+
+        if not self.add_bos_token:
+            return super().get_special_tokens_mask(
+                token_ids_0=token_ids_0, token_ids_1=token_ids_1, already_has_special_tokens=False
+            )
+
+        if token_ids_1 is None:
```

Comment:
Do you think this should only be modified here and not in the `tokenization_base` class? I wonder if it could be affecting other models but silently, but my knowledge of the tokenizer is clearly lacking!

Reply:
Yes I think it should be changed here, the other tokenizers that add special tokens override the method `get_special_tokens_mask` defined in `tokenization_base` class. :relaxed: 

For example:
https://github.com/huggingface/transformers/blob/693ba2cc79ef46e8323b35f791233ce94f433b5d/src/transformers/models/bert/tokenization_bert.py#L293-L319